model/schema.json

```diff
@@ -607,6 +607,202 @@
         "api_key_id",
         "api_key"
       ]
+    },
+    "filechunk": {
+      "title": "Chunk",
+      "description": "One section of the blob contents of a file",
+      "type": "object",
+      "properties": {
+        "_id": {
+          "description": "The identifier for the file chunk, describing the ID of the file, and the chunk position",
+          "type": "string"
+        },
+        "data": {
+          "description": "Blob contents of the file",
+          "type": "binary"
+        },
+        "bid": {
+          "description": "Base ID of the chunk, matching all chunks together as the base ID of the file",
+          "type": "string",
+          "format": "uuid"
+        },
+        "last": {
+          "description": "Whether this chunk is the final chunk of a file",
+          "type": "boolean"
+        }
+      }
+    },
+    "file-info": {
+      "title": "File Info",
+      "description": "An uploaded File",
+      "type": "object",
+      "properties": {
+        "_id": {
+          "description": "The unique identifier for the file metadata",
+          "type": "string",
+          "format": "uuid"
+        },
+        "agent_id": {
+          "description": "The unique identifier of the uploading agent",
+          "type": "string",
+          "format": "uuid"
+        },
+        "action_id": {
+          "description": "The identifier for the action requesting this upload",
+          "type": "string",
+          "format": "uuid"
+        },
+        "source": {
+          "description": "The integration initiating this file upload",
+          "type": "string"
+        },
+        "file": { "$ref": "#/definitions/file-data" },
+        "contents": {
+          "description": "Description of the contents of an archive, when the file property describes an archive (e.g. zip)",
+          "type": "array",
+          "items": {
+            "ref": "#/definitions/file-data"
+          }
+        },
```

Comment:
I have yet to read the proposal so I may be lacking in knowledge here. But is `contents` required for an archive? or just recommended?

To give some context: my use case for these endpoints will be having the `elastic-agent` upload a diagnostics bundle (zip file).

Reply:
contents is not required. It will be used by Endpoint since the facts surrounding the file _within_ the zip may be very relevant to an investigation. For agent, unless one wants to know the hash or details or contents of the archive, it is very much not required.

If multiple files were present within the archive, and one wanted to display that fact in a UI on kibana (e.g. `"Agent Diagnostics 2022-10-20  (3 files)"`) then it could be used for that purpose.